content/open-source/wolfi/wolfi-with-dockerfiles.md

```diff
@@ -0,0 +1,131 @@
+
+To test that the application works as expected, first install the dependencies using Pip.
+
+```shell
+pip3 install -r requirements.txt
+```
+
+Then, run the application with:
+
+```shell
+python3 inky.py
+```
+
+You’ll receive a representation of the Chainguard Inky logo on the command line. With the demo application ready, you can now move on to building the Wolfi container image.
+
+## Creating the Dockerfile
+
+Now we'll create the Dockerfile to run the application. This Dockerfile will set up a new WORKDIR, copy relevant files, and install dependencies with Pip. It will also define the entry point that will be executed when we run this image with `docker run`.
+
+You can rename the old Dockerfile if you want to keep it for tests later.
+
+```shell
+mv Dockerfile _DockerfileBkp
+```
+
+Then, create a new Dockerfile:
+
+```shell
+nano Dockerfile
+```
+
+Copy the following content to it:
+
+```
+FROM cgr.dev/chainguard/wolfi-base
+
+WORKDIR /app
+COPY requirements.txt inky.py inky.png /app/
+RUN apk add python-3.11 py3.11-pip && \
+    pip3 install -r requirements.txt --user
+
+ENTRYPOINT ["python", "/app/inky.py"]
+
+```
+
+After saving the file, you can build your application runtime image with:
+
+```shell
+docker build . -t inky-demo
+```
+
+Finally, run the image with:
+
+```shell
+docker run --rm inky-demo
+```
+
+And you should get a similar result as the previous time you run the script.
+
+## Building distroless Wolfi Images
+
+A _distroless_ image is typically a very minimal container image that doesn't have shells or package managers. The extra tightness improves security in several aspects, but it requires a more sophisticated strategy for image composition since you can't install packages so easily. There are currently two main strategies for building distroless images with Wolfi:
+
+- **With a Dockerfile:** Use `-dev` variants from [Chainguard Images](https://edu.chainguard.dev/chainguard/chainguard-images/overview/) to build the application, and copy the artifacts to a pure distroless image in a [Docker multi-stage build](https://docs.docker.com/build/building/multi-stage/). This option is typically more accessible for people who are already used to a Dockerfile workflow. Check the [Getting Started with the Python Chainguard Image](https://edu.chainguard.dev/chainguard/chainguard-images/reference/python/getting-started-python/) guide for practical examples.
+- **With apko:** Use [apko](https://edu.chainguard.dev/open-source/apko/overview/) to build a distroless image with only the packages you need, fully customized. This option requires a bigger learning curve to get used to how apko works, but it will give you smaller images with a better SBOM coverage. Check the [Getting Started with apko](https://edu.chainguard.dev/open-source/apko/getting-started-with-apko/) tutorial to see how that works in practice.
```

Comment:
```suggestion
- **With apko:** Use [apko](https://edu.chainguard.dev/open-source/apko/overview/) to build a distroless image with only the packages you need, fully customized. This option requires a bigger learning curve to get used to how apko works, but it will give you smaller images with better SBOM coverage. Check the [Getting Started with apko](https://edu.chainguard.dev/open-source/apko/getting-started-with-apko/) tutorial to understand how that works in practice.
```